pandas/core/arrays/datetimes.py

```diff
@@ -66,6 +74,50 @@ class DatetimeArrayMixin(DatetimeLikeArrayMixin):
                  'is_year_end', 'is_leap_year']
     _object_ops = ['weekday_name', 'freq', 'tz']
 
+    # -----------------------------------------------------------------
+    # Constructors
+
+    _attributes = ["freq", "tz"]
+
+    @classmethod
+    def _simple_new(cls, values, freq=None, tz=None, **kwargs):
+        """
+        we require the we have a dtype compat for the values
+        if we are passed a non-dtype compat, then coerce using the constructor
+        """
+
+        if getattr(values, 'dtype', None) is None:
+            # empty, but with dtype compat
+            if values is None:
+                values = np.empty(0, dtype=_NS_DTYPE)
+                return cls(values, freq=freq, tz=tz, **kwargs)
+            values = np.array(values, copy=False)
+
+        if not is_datetime64_dtype(values):
+            values = _ensure_int64(values).view(_NS_DTYPE)
+
+        result = object.__new__(cls)
+        result._data = values
+        result._freq = freq
+        tz = timezones.maybe_get_tz(tz)
+        result._tz = timezones.tz_standardize(tz)
+        return result
+
+    def __new__(cls, values, freq=None, tz=None):
+        if (freq is not None and not isinstance(freq, DateOffset) and
```

Comment:
this condition could actually be handled to by ``to_offset`` (future PR) (except for the not-None part)